src/install.ts

```diff
@@ -0,0 +1,119 @@
+// Automatically install clangd binary releases from GitHub.
+// This wraps `@clangd/install` in the VSCode UI. See that package for more.
+
+import * as common from '@clangd/install';
+import AbortController from 'abort-controller';
+import * as vscode from 'vscode';
+
+// Returns the clangd path to be used, or null if clangd is not installed.
+export async function activate(context: vscode.ExtensionContext):
+    Promise<string> {
+  const cfg = vscode.workspace.getConfiguration('clangd');
+  const ui = new UI(context, cfg);
+  context.subscriptions.push(vscode.commands.registerCommand(
+      'clangd.install', async () => common.installLatest(ui)));
+  context.subscriptions.push(vscode.commands.registerCommand(
+      'clangd.update', async () => common.checkUpdates(true, ui)));
+  const status = await common.prepare(ui, cfg.get<boolean>('checkUpdates'));
+  return status.clangdPath;
+}
+
+class UI {
+  constructor(private context: vscode.ExtensionContext,
```

Comment:
I was going to say the parameters are unused, then realized the TS compiler will generate the class members for us.

Reply:
Yeah, it's weird syntax but pretty handy...